lib/tasks/create_fake_data.rake

```diff
@@ -0,0 +1,89 @@
+namespace :db do
+  namespace :seed do
+    desc 'Generate fake patient entries for data wranglers'
+    task :create_fake_data => :environment do
+
+        users = User.all
+        clinics = Clinic.all
+
+        5000.times do |idx|
+          flag = (rand(10) % 5 == 0)
+
+          initial_call = Date.today - rand(1000)
+          
+          has_appt = rand(3) % 2 == 0
+
+          has_pledge = rand(3) != 2 
+
+          lines = ['DC', 'VA', 'MD'] # need to add Spanish maybe? 
+
+          patient = Patient.create!(
+            name: 'Randomized Patient',
+            primary_phone: SecureRandom.random_number(10**10).to_s.rjust(10, '0'), 
+            initial_call_date: initial_call,
+            created_by: users.sample(1).first,
+            urgent_flag: flag,
+            line: lines[idx%3], # thank you seeds.rb! 
+            clinic: has_appt ? clinics.sample(1).first : nil,
+            appointment_date: has_appt ? initial_call + rand(30) : nil,
+            last_menstrual_period_weeks: rand(15) + 7,
+            last_menstrual_period_days: rand(7),
+            procedure_cost: has_appt ? rand(600) : nil,
+            pledge_sent: has_appt && has_pledge,
+            patient_contribution: rand(400),
+            fund_pledge: has_appt ? rand(300) : nil
+          )
+
+          # create external_pledges
+          if idx.odd? && has_appt
+            patient.external_pledges.create!(
+            source: 'Metallica Abortion Fund',
+            amount: 100,
+            created_by: User.first
+            )
+          end
+
+          # create calls
+          call_status = ["Left voicemail", "Reached patient", "Couldn't reach patient"]
+
+          if idx.odd? && !has_appt
```

Comment:
I'm a little confused by the conditionals. Will the second one always happen, or will there be times when people sometimes have no calls?

It might be worth tweaking for clarity, or adding a comment about how this works.

Reply:
Yes, I was initially thinking that some people will have no calls and the conditionals was kind of a lazy way of randomizing that. Thinking about it further now, it's more likely that people will have at least one call once they're entered in DARIA.